runtime/common/src/claims.rs

```diff
@@ -161,31 +161,23 @@ decl_module! {
 			let balance_due = <Claims<T>>::get(&signer)
 				.ok_or(Error::<T>::SignerHasNoClaim)?;
 
-			let maybe_vested = <Vesting<T>>::get(&signer);
-
-			// If this fails, destination account already has a vesting schedule
-			// applied to it, and this claim should not be processed.
-			ensure!(
```

Comment:
When you removed this ensure check, I also believe you now made it possible for someone with a vesting schedule to drop the pot to zero by calling this function over and over.

Reply:
I have updated the logic and added a test which would have failed if this logic didn't exist.

https://github.com/paritytech/polkadot/pull/834/commits/15fe7cc4008582cd630fa564204115210c5de258